### PR TITLE
RDCC-3168: Addressing Sonar Issues in Master

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-3168

### Change description ###

Addressing `Sonar` Issues in Master

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java [hunk 1]
```diff
@@ -66,40 +66,19 @@ public void test_Decode_without_gzip() {
...
+                JsonFeignResponseHelper.decode(response1, Optional.of(UserProfileCreationResponse.class));
 
-        Response response = Response.builder().status(400).reason("OK").headers(header)
-                .body("{\"idamId\": 1}", UTF_8).request(request).build();
+        assertThat(createUserProfileResponseOptional1).isNotEmpty();
 
-        Optional<UserProfileCreationResponse> createUserProfileResponseOptional = JsonFeignResponseHelper
-                .decode(response, Optional.of(UserProfileCreationResponse.class));
+        Optional<UserProfileCreationResponse> createUserProfileResponseOptional2 = JsonFeignResponseHelper
+                .decode(response1, Optional.of(UserProfileCreationResponse.class));
```
Comment (ishratt) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java:79]: are we expecting any differences in response between `createUserProfileResponseOptional1` and `createUserProfileResponseOptional2` as both are results of `JsonFeignResponseHelper.decode(response1, Optional.of(UserProfileCreationResponse.class))`?

**Review by kavithagade12 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java (hunk 1 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java:79]: In one of them the `response` status is `200` and in the other the status is `400`.

_Please note that I did not write these tests, I have merely combined them into 1 test as per Sonar_

Comment (ishratt) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java:79]: Thanks for your reply. Looking at the code before the latest changes, it seems like we had some code duplication in the tests (same code just different method names). As we are refactoring the tests here, I believe we can remove lines `78-81` as lines `73-76` are doing the same, but correct me if I'm wrong.

Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/util/JsonFeignResponseHelperTest.java]: Addressed here: https://github.com/hmcts/rd-user-profile-api/pull/721/commits/e79edabc0d9d0a5a84e174d2a4b4e59b4a80d485

**Review by ishratt — APPROVED** [review]

**Review by rajkatla-hmcts — APPROVED** [review]

**Pull request merged**